spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreEntityConverterImplTests.java

```diff
@@ -0,0 +1,167 @@
+		TestDatastoreItem item = entityConverter.read(TestDatastoreItem.class, entity);
+
+		assertThat(item.getStringField()).as("validate null field").isNull();
+	}
+
+	@Test(expected = DatastoreDataException.class)
+	public void readTestException() {
+				Entity entity = Entity.newBuilder(this.datastore.newKeyFactory().setKind("aKind").newKey("1"))
+				.set("stringField", "string value")
+				.set("boolField", 123L)
+				.build();
+
+		DatastoreEntityConverter entityConverter = new DatastoreEntityConverterImpl(new DatastoreMappingContext());
+		entityConverter.read(TestDatastoreItem.class, entity);
+	}
+
+	@Test
+	public void writeTest() {
+		byte[] bytes = { 1, 2, 3 };
+		TestDatastoreItem item = new TestDatastoreItem();
+		item.setStringField("string value");
+		item.setBoolField(true);
+		item.setDoubleField(3.1415D);
+		item.setLongField(123L);
+		item.setLatLngField(LatLng.of(10, 20));
+		item.setTimestampField(Timestamp.ofTimeSecondsAndNanos(30, 40));
+		item.setBlobField(Blob.copyFrom(bytes));
+
+		DatastoreEntityConverter entityConverter = new DatastoreEntityConverterImpl(new DatastoreMappingContext());
+		Entity.Builder builder = Entity.newBuilder(this.datastore.newKeyFactory().setKind("aKind").newKey("1"));
+		entityConverter.write(item, builder);
+
+		Entity entity = builder.build();
+
+		assertThat(entity.getString("stringField")).as("validate string field").isEqualTo("string value");
+		assertThat(entity.getBoolean("boolField")).as("validate boolean field").isTrue();
+		assertThat(entity.getDouble("doubleField")).as("validate double field").isEqualTo(3.1415D);
+		assertThat(entity.getLong("longField")).as("validate long field").isEqualTo(123L);
+		assertThat(entity.getLatLng("latLngField")).as("validate latLng field").isEqualTo(LatLng.of(10, 20));
+		assertThat(entity.getTimestamp("timestampField")).as("validate timestamp field")
+				.isEqualTo(Timestamp.ofTimeSecondsAndNanos(30, 40));
+		assertThat(entity.getBlob("blobField")).as("validate blob field").isEqualTo(Blob.copyFrom(bytes));
+	}
+
+	@Test
+	public void writeNullTest() {
+		byte[] bytes = { 1, 2, 3 };
+		TestDatastoreItem item = new TestDatastoreItem();
+		item.setStringField(null);
+		item.setBoolField(true);
+		item.setDoubleField(3.1415D);
+		item.setLongField(123L);
+		item.setLatLngField(LatLng.of(10, 20));
+		item.setTimestampField(Timestamp.ofTimeSecondsAndNanos(30, 40));
+		item.setBlobField(Blob.copyFrom(bytes));
+
+		DatastoreEntityConverter entityConverter = new DatastoreEntityConverterImpl(new DatastoreMappingContext());
+		Entity.Builder builder = Entity.newBuilder(this.datastore.newKeyFactory().setKind("aKind").newKey("1"));
+		entityConverter.write(item, builder);
+
+		Entity entity = builder.build();
+
+		assertThat(entity.getValue("stringField").equals(new NullValue()))
+				.as("validate null field").isTrue();
+	}
+
+	@Test(expected = DatastoreDataException.class)
+	public void writeTestException() {
```

Comment:
This could have a more descriptive name about specifically what is wrong that is causing the exception.